packages/backend/apps/multitenancy/models.py

```diff
@@ -21,14 +44,27 @@ class Tenant(TimestampedMixin, models.Model):
 
     objects = TenantManager()
 
+    __original_name = None
+
     def __str__(self):
         return self.name
 
     def save(self, *args, **kwargs):
```

Comment:
I might be wrong but proper approach is to try to insert into db immediately. If there's an error, increase counter and try to insert into DB again. Whole "Ask for forgiveness, not for permission" thing

Your current implementation is susceptible to phantom read (or is that non-repeatable ready?) anomaly.
Unique constraint will protect db but if two users create with the same name at the same time one of them will just get a 500 error.